src/lib/components/Accordion/AccordionItem.svelte

```diff
@@ -0,0 +1,52 @@
+<script lang="ts">
+  import { getContext, onMount } from "svelte";
+  import type { AccordionContext, AccordionContextItems } from "./types";
+
+  // Note: `id` has a fallback value, but ideally,
+  // one should supply their own identifiers
+  export let id = "accordionItem-" + Math.random().toString(36);
+  export let title: string | null | undefined = "Title";
+  export let expanded = false;
+
+  const context: AccordionContext = getContext("Accordion");
+
+  let ref: HTMLElement | null = null;
+  let unsubscribe: () => void | undefined;
+
+  onMount(() => {
+    return () => {
+      if (context) context.remove({ id, expanded: !expanded });
+      if (unsubscribe) unsubscribe();
+    };
+  });
+
+  $: if (context) {
+    context.add({ id, expanded });
+    unsubscribe = context.items.subscribe((value: AccordionContextItems) => {
+      expanded = value[id];
+    });
+  }
```

Comment:
Looking over this again, this seems like a danger of adapting others' code. This feels a bit convoluted and that I should just refactor this to use a store. Context, as I understand it, isn't reactive, whereas store _is_ and is meant to be subscribed to.

Reply:
So basically doing what [this article](https://joyofcode.xyz/svelte-context-with-stores) describes and using context to pass a store to descendant components. Then you get the clean API of stores with dealing with the global scope of a store